test/unit-test/FreeRTOS_IPv6_Utils/FreeRTOS_IPv6_Utils_utest.c

```diff
@@ -0,0 +1,359 @@
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_PING_REPLY_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( usReturn, 10 );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function verify sending an valid length
+ *        for ipICMP_PING_REPLY_IPv6.
+ */
+void test_prvChecksumICMPv6Checks_PingReply_ValidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPEcho_IPv6_t ) + 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_PING_REPLY_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( usReturn, pdFALSE );
+    TEST_ASSERT_EQUAL( sizeof( ICMPEcho_IPv6_t ), xSet.uxProtocolHeaderLength );
+}
+
+/**
+ * @brief This function verify sending an invalid length
+ *        for ipICMP_ROUTER_SOLICITATION_IPv6.
+ */
+void test_prvChecksumICMPv6Checks_RS_InvalidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPRouterSolicitation_IPv6_t ) - 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_ROUTER_SOLICITATION_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( usReturn, 10 );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function verify sending an valid length
+ *        for ipICMP_ROUTER_SOLICITATION_IPv6.
+ */
+void test_prvChecksumICMPv6Checks_RS_ValidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPRouterSolicitation_IPv6_t ) + 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_ROUTER_SOLICITATION_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( usReturn, pdFALSE );
```

Comment:
```suggestion
    TEST_ASSERT_EQUAL( 0, usReturn );
```
I suggest to correct the value&sequence here.
The first input is expected value, and the second one is the variable to be examined

Reply:
Updated